### DISCUSSION TITLE
AutoGluon with Chronos_Bolt Generates Empty Folders During Time Series Forecasting – Code Issue?

### DISCUSSION THREAD

**Discussion by focus1989** [autogluon/autogluon#5108] [General]
"I'm using AutoGluon with Chronos_Bolt for time series load forecasting, and GBM for covariate prediction. However, it's generating many empty folders with names like 'ag-20250430_013524LGBModel'. Is there something wrong with my code? Can someone help me? Below is the code:"
predictor = TimeSeriesPredictor(
    prediction_length=prediction_length,
    target="load",
    known_covariates_names=[
         "tempreture", "Irradiance", "Wind_Speed", "Wind_Direction", "Humidity", "Pressure", "Holidays","Weekend"
    ],
    eval_metric="WQL",
    # quantile_levels=[0.8, 0.9],
    freq="15min",
    verbosity = 4,
).fit(
    train_data,
    hyperparameters={
        "Chronos": [
            {

                "fine_tune_lr" : 1e-5,
                "model_path": "bolt_small",
                "context_length": 1344,
                "fine_tune " :True, 
                "ag_args":{"name_suffix": "FineTuned"}
            },
            {
                # "fine_tune" : True,
                "fine_tune_lr" : 1e-5,
                "model_path": "bolt_small",
                "covariate_regressor": "GBM",
                "target_scaler": "standard",
                "ag_args": {"name_suffix": "WithRegressor"},
                "context_length": 1344,
            },
            {
                # "fine_tune" : True,
                "fine_tune_lr" : 1e-5,
                "model_path": "bolt_tiny",
                "covariate_regressor": "GBM",
                "target_scaler": "standard",
                "ag_args": {"name_suffix": "WithRegressor"},
                "context_length": 1344,
            },
            {
                # "fine_tune" : True,
                "fine_tune_lr" : 1e-5,
                "model_path": "bolt_mini",
                "covariate_regressor": "GBM",
                "target_scaler": "standard",
                "ag_args": {"name_suffix": "WithRegressor"},
                "context_length": 1344,
            },
            {
                # "fine_tune" : True,
                "fine_tune_lr" : 1e-5,
                "model_path": "bolt_base",
                "covariate_regressor": "GBM",
                "target_scaler": "standard",
                "ag_args": {"name_suffix": "WithRegressor"},
                "context_length": 1344,
            },
        ],
    },
    # hyperparameter_tune_kwargs="auto",
    enable_ensemble=True,
    time_limit=1800,
    num_val_windows=80,
    presets="best_quality"
)

**Comment by focus1989** [comment]
this is predictor_log
Beginning AutoGluon training... Time limit = 1800s
AutoGluon will save models to '/home/featurize/work/autoglun/AutogluonModels/ag-20250430_031324'
=================== System Info ===================
AutoGluon Version:  1.2
Python Version:     3.11.11
Operating System:   Linux
Platform Machine:   x86_64
Platform Version:   #115-Ubuntu SMP Mon Apr 15 09:52:04 UTC 2024
CPU Count:          16
GPU Count:          1
Memory Avail:       51.83 GB / 54.92 GB (94.4%)
Disk Space Avail:   26.51 GB / 50.00 GB (53.0%)
===================================================
Setting presets to: best_quality

Fitting with arguments:
{'enable_ensemble': True,
 'eval_metric': WQL,
 'freq': '15min',
 'hyperparameters': {'Chronos': [{'ag_args': {'name_suffix': 'FineTuned'},
                                  'context_length': 1344,
                                  'fine_tune ': True,
                                  'fine_tune_lr': 1e-05,
                                  'model_path': 'bolt_small'},
                                 {'ag_args': {'name_suffix': 'WithRegressor'},
                                  'context_length': 1344,
                                  'covariate_regressor': 'GBM',
                                  'fine_tune_lr': 1e-05,
                                  'model_path': 'bolt_small',
                                  'target_scaler': 'standard'},
                                 {'ag_args': {'name_suffix': 'WithRegressor'},
                                  'context_length': 1344,
                                  'covariate_regressor': 'GBM',
                                  'fine_tune_lr': 1e-05,
                                  'model_path': 'bolt_tiny',
                                  'target_scaler': 'standard'},
                                 {'ag_args': {'name_suffix': 'WithRegressor'},
                                  'context_length': 1344,
                                  'covariate_regressor': 'GBM',
                                  'fine_tune_lr': 1e-05,
                                  'model_path': 'bolt_mini',
                                  'target_scaler': 'standard'},
                                 {'ag_args': {'name_suffix': 'WithRegressor'},
                                  'context_length': 1344,
                                  'covariate_regressor': 'GBM',
                                  'fine_tune_lr': 1e-05,
                                  'model_path': 'bolt_base',
                                  'target_scaler': 'standard'}]},
 'known_covariates_names': ['温度', '辐照度', '风速', '风向', '湿度', '压力', '节假日', '周末'],
 'num_val_windows': 2,
 'prediction_length': 192,
 'quantile_levels': [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9],
 'random_seed': 123,
 'refit_every_n_windows': 1,
 'refit_full': False,
 'skip_model_selection': False,
 'target': '变压器总负荷',
 'time_limit': 1800,
 'verbosity': 4}

Provided train_data has 735648 rows, 83 time series. Median time series length is 8928 (min=3552, max=8928). 

Provided data contains following columns:
	target: '变压器总负荷'
	known_covariates:
		categorical:        ['温度', '辐照度', '风速', '湿度', '压力']
		continuous (float): ['节假日', '周末']
	past_covariates:
		categorical:        []
		continuous (float): ['变压器下大用户负荷', '变压器下发电负荷', '储能', '光伏', '电厂', '风电', ...]
	static_features:
		categorical:        []
		continuous (float): ['cn_level', 'gf_level', 'dc_level', 'wd_level']

AutoGluon will ignore following non-numeric/non-informative columns:
	ignored covariates:      ['风向']

To learn how to fix incorrectly inferred types, please see documentation for TimeSeriesPredictor.fit

AutoGluon will gauge predictive performance using evaluation metric: 'WQL'
	This metric's sign has been flipped to adhere to being higher_is_better. The metric score can be multiplied by -1 to get the metric value.
===================================================

Starting training. Start time is 2025-04-30 03:13:37
Models that will be trained: ['ChronosFineTuned[bolt_small]', 'ChronosWithRegressor[bolt_small]', 'ChronosWithRegressor[bolt_tiny]', 'ChronosWithRegressor[bolt_mini]', 'ChronosWithRegressor[bolt_base]']
Reserving 299.3s for ensemble
Training timeseries model ChronosFineTuned[bolt_small]. Training for up to 299.3s of the 1795.8s of remaining time.
	Window 0
		-0.2238      = Validation score (-WQL)
		20.996  s    = Training runtime
		2.429   s    = Prediction runtime
	Window 1
		-0.2344      = Validation score (-WQL)
		0.005   s    = Training runtime
		1.494   s    = Prediction runtime
	-0.2291       = Validation score (-WQL)
	24.41   s     = Training runtime
	1.49    s     = Validation (prediction) runtime
Reserving 354.0s for ensemble
Training timeseries model ChronosWithRegressor[bolt_small]. Training for up to 354.0s of the 1769.8s of remaining time.
	Window 0
		-0.2186      = Validation score (-WQL)
		8.226   s    = Training runtime
		1.950   s    = Prediction runtime
	Window 1
		-0.2381      = Validation score (-WQL)
		7.522   s    = Training runtime
		3.054   s    = Prediction runtime
	-0.2283       = Validation score (-WQL)
	18.64   s     = Training runtime
	3.05    s     = Validation (prediction) runtime
Reserving 437.0s for ensemble
Training timeseries model ChronosWithRegressor[bolt_tiny]. Training for up to 437.0s of the 1748.1s of remaining time.
	Window 0
		-0.2189      = Validation score (-WQL)
		9.941   s    = Training runtime
		2.817   s    = Prediction runtime
	Window 1
		-0.2404      = Validation score (-WQL)
		7.109   s    = Training runtime
		1.976   s    = Prediction runtime
	-0.2296       = Validation score (-WQL)
	20.80   s     = Training runtime
	1.98    s     = Validation (prediction) runtime
Reserving 575.1s for ensemble
Training timeseries model ChronosWithRegressor[bolt_mini]. Training for up to 575.1s of the 1725.3s of remaining time.
	Window 0
		-0.2066      = Validation score (-WQL)
		9.244   s    = Training runtime
		2.260   s    = Prediction runtime
	Window 1
		-0.2348      = Validation score (-WQL)
		7.363   s    = Training runtime
		2.321   s    = Prediction runtime
	-0.2207       = Validation score (-WQL)
	19.80   s     = Training runtime
	2.32    s     = Validation (prediction) runtime
Reserving 600.0s for ensemble
Training timeseries model ChronosWithRegressor[bolt_base]. Training for up to 1103.1s of the 1703.1s of remaining time.
	Window 0
		-0.1987      = Validation score (-WQL)
		10.333  s    = Training runtime
		2.345   s    = Prediction runtime
	Window 1
		-0.2278      = Validation score (-WQL)
		7.460   s    = Training runtime
		2.513   s    = Prediction runtime
	-0.2133       = Validation score (-WQL)
	21.13   s     = Training runtime
	2.51    s     = Validation (prediction) runtime
Fitting simple weighted ensemble.
	Ensemble weights: {'ChronosWithRegressor[bolt_base]': 1.0}
	-0.2133       = Validation score (-WQL)
	2.49    s     = Training runtime
	2.51    s     = Validation (prediction) runtime
Training complete. Models trained: ['ChronosFineTuned[bolt_small]', 'ChronosWithRegressor[bolt_small]', 'ChronosWithRegressor[bolt_tiny]', 'ChronosWithRegressor[bolt_mini]', 'ChronosWithRegressor[bolt_base]', 'WeightedEnsemble']
Total runtime: 119.71 s
Best model: ChronosWithRegressor[bolt_base]
Best model score: -0.2133
Generating leaderboard for all models trained
Additional data provided, testing on additional data. Resulting leaderboard will be sorted according to test score (`score_test`).
Found no cached predictions
Prediction order: ['ChronosFineTuned[bolt_small]', 'ChronosWithRegressor[bolt_mini]', 'ChronosWithRegressor[bolt_small]', 'ChronosWithRegressor[bolt_tiny]', 'ChronosWithRegressor[bolt_base]', 'WeightedEnsemble']
Cached predictions saved to /home/featurize/work/autoglun/AutogluonModels/ag-20250430_031324/models/cached_predictions.pkl
Model not specified in predict, will default to the model with the best validation score: ChronosWithRegressor[bolt_base]
Loaded cached predictions for models ['ChronosFineTuned[bolt_small]', 'ChronosWithRegressor[bolt_mini]', 'ChronosWithRegressor[bolt_small]', 'ChronosWithRegressor[bolt_tiny]', 'ChronosWithRegressor[bolt_base]', 'WeightedEnsemble']
Prediction order: {'ChronosWithRegressor[bolt_base]'}
Extending existing cached predictions
Cached predictions saved to /home/featurize/work/autoglun/AutogluonModels/ag-20250430_031324/models/cached_predictions.pkl

**Reply by shchur** [reply]
This bug is fixed in the latest release (v1.3). Feel free to reopen the issue if there problem persists
